docs/source/common_issues.rst

```diff
@@ -51,6 +51,12 @@ flagged as an error.
 
   If you don't know what types to add, you can use ``Any``, but beware:
 
+- **A function annotated as non-optional returns ``None`` and mypy doesn't
+  complain** By default, ``None`` value is considered compatible
```

Comment:
"`None` value" > "`None`" or put "the" before `None`, since you are talking about a definite noun.